staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -115,6 +117,9 @@ type Config struct {
 	EnableMetrics             bool
 
 	DisabledPostStartHooks sets.String
+	postStartHookLock      sync.Mutex
```

Comment:
I don't like this lock in a config. Configs are always non-threaded. Why should this be different.